src/index.js

```diff
@@ -272,8 +272,12 @@ export default class Critters {
         rule.selectors = rule.selectors.filter(sel => {
           // Strip pseudo-elements and pseudo-classes, since we only care that their associated elements exist.
           // This means any selector for a pseudo-element or having a pseudo-class will be inlined if the rest of the selector matches.
-          sel = sel.replace(/::?(?:[a-z-]+)([.[#~&^:*]|\s|\n|$)/gi, '$1');
-          return document.querySelector(sel, document) != null;
+          sel = sel.replace(/::?(?:[a-z-]+)([.[#~&^:*]|\s|\n|$)/gi, '$1').trim();
+          try {
+            return document.querySelector(sel, document) != null;
+          } catch (e) {
+            return null;
```

Comment:
@developit, I guess we can add a string telling which selector failed here

Reply:
that'd be good!